src/plugins/share/README.mdx

```diff
@@ -0,0 +1,243 @@
+---
+id: kibDevSharePluginReadme
+slug: /kibana-dev-docs/tutorials/share
+title: Kibana share Plugin
+summary: Introduction to Kibana "share" plugin services (sharing popup menu, URL locators, and short URLs).
+date: 2022-02-21
+tags: ['kibana', 'onboarding', 'dev', 'tutorials', 'share', 'url', 'short-url', 'locator']
+---
+
+# Share plugin
+
+The `share` plugin contains various utilities for displaying sharing context menu,
+generating deep links to other apps using *locators*, and creating short URLs.
+
+
+## Sharing context menu
+
+You can register an item into sharing context menu (which is displayed in
+Dashboard, Discover, and Visualize apps).
+
+### Example registration
+
+```ts
+import { ShareContext, ShareMenuItem } from 'src/plugins/share/public';
+
+plugins.share.register({
+  id: 'demo',
+  getShareMenuItems: (context) => [
+    {
+      panel: {
+        id: 'demo',
+        title: 'Panel title',
+        content: 'Panel content',
+      },
+      shareMenuItem: {
+        name: 'Demo list item (from share_example plugin)',
+      },
+    }
+  ],
+});
+```
+
+Now the "Demo list item" will appear under the "Share" menu in Discover and
+Dashboard applications.
```

Comment:
worth mentioning how to get it to display in other applications ? (how to add that menu to your app)